src/vs/editor/contrib/gotoSymbol/peek/referencesController.ts

```diff
@@ -278,6 +290,30 @@ function withController(accessor: ServicesAccessor, fn: (controller: ReferencesC
 	}
 }
 
+KeybindingsRegistry.registerCommandAndKeybindingRule({
+	id: 'changeFocus',
+	weight: KeybindingWeight.WorkbenchContrib + 50,
+	primary: KeyCode.F2,
+	when: ctxReferenceSearchVisible,
+	handler(accessor) {
+		withController(accessor, controller => {
+			controller.changeFocusBetweenPreviewAndReferences();
+		});
+	}
+});
+
+KeybindingsRegistry.registerCommandAndKeybindingRule({
+	id: 'changeFocusFromEmbeddedEditor',
+	weight: KeybindingWeight.EditorContrib + 50,
+	primary: KeyCode.F2,
+	when: PeekContext.inPeekEditor,
```

Comment:
needs a try but I believe that this isn't needed anymore now that `when` supports OR

Reply:
If KeyCode.F2 is set to primary then '_when_' conditions **correctly appear** on Keyboard Shortcuts screen:

![image](https://user-images.githubusercontent.com/10179520/70007074-d6491400-154d-11ea-8010-38f989986adc.png)

But if I remove KeyCode.F2 or change it to KeyCode.Unknown, '_when_' conditions **do not appear** anymore.

![image](https://user-images.githubusercontent.com/10179520/70007152-16a89200-154e-11ea-961e-361601f1f32d.png)

Should they appear? Is it a bug? Because if they really shouldn't appear I could simply remove '_when_' clause from this new command.

Reply:
On the other hand, the _when_ clause is highly associated with the specified KeyBinding. If no KeyBinding is set on 'primary' option, then I think it makes sense to be empty as well...

Maybe I just should set _when_ clause to undefined or something similar.

Reply:
yeah, the when-clause only comes to use with a keybinding. I think it's OK to add a keybinding but not one that is in use already